src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,179 +1,248 @@
+                        toolTip.add("")
+                        toolTip.add("§cMissing: §7" + StringUtils.createCommaSeparatedList(classNames))
+                    }
+                    if(toolTip.isNotEmpty()) toolTipMap[slot] = toolTip
                 }
             }
+        }
 
-            val members = slot.stack.getLore().filter { memberPattern.matches(it) }
-            val memberLevels = members.map { memberPattern.matchEntire(it)?.groupValues?.get(2)?.toInt() ?: 0 }
-            val memberClasses = members.map { memberPattern.matchEntire(it)?.groupValues?.get(1) ?: "" }
 
-            if (memberLevels.any { it <= config.markBelowClassLevel }) {
-                slot highlight LorenzColor.YELLOW
-                continue
-            }
+    }
 
-            if (config.markMissingClass && memberClasses.none { it == selectedClass }) {
-                slot highlight LorenzColor.GREEN
-            }
+    @SubscribeEvent
+    fun onToolTipRender(event: LorenzToolTipEvent) {
+        if (!isEnabled()) return
+        if (!inInventory) return
+        val toolTip = toolTipMap[event.slot.slotIndex]
+        if (toolTip.isNullOrEmpty()) return
+        toolTip.withIndex().forEach { (index, line) ->
+            if(index >= event.toolTip.size) event.toolTip.add(line)
+                else event.toolTip[index] = line
         }
     }
 
     @SubscribeEvent
-    fun onItemTooltip(event: LorenzToolTipEvent) {
-        if (!LorenzUtils.inSkyBlock) return
-
-        val chestName = InventoryUtils.openInventoryName()
-        if (chestName != "Party Finder") return
-
-        val stack = event.itemStack
-
-        val classNames = mutableListOf("Healer", "Mage", "Berserk", "Archer", "Tank")
-        for ((index, line) in stack.getLore().withIndex()) {
-            classLevelPattern.matchMatcher(line) {
-                val playerName = group("playerName")
-                val className = group("className")
-                val level = group("level").toInt()
-                val color = getColor(level)
-                if (config.coloredClassLevel) event.toolTip[index + 1] = " §b$playerName§f: §e$className $color$level"
-                classNames.remove(className)
-            }
-        }
-        if (!config.showMissingClasses) return
-        if (stack.getLore().firstOrNull()?.removeColor()?.startsWith("Dungeon:") == false) return
-        if (classNames.contains(selectedClass)) selectedClass = "§a${selectedClass}§7"
-        event.toolTip.add("")
-        event.toolTip.add("§cMissing: §7" + createCommaSeparatedList(classNames))
+    fun onRenderItemTip(event: RenderInventoryItemTipEvent) {
+        if(!isEnabled()) return
+        if(!config.floorAsStackSize) return
+        if(floorStackSize[event.slot.slotIndex].isNullOrEmpty()) return
+        event.stackTip = floorStackSize[event.slot.slotIndex].toString()
```

Comment:
if null, empty String
```suggestion
        event.stackTip = floorStackSize[event.slot.slotIndex] ?: ""
```

Reply:
Also formatting